src/sidebar/store/modules/annotations.js

```diff
@@ -437,6 +442,25 @@ function findAnnotationByID(state, id) {
   return findByID(state.annotations.annotations, id);
 }
 
+/**
+ * Return all loaded annotations which have not been saved to the server.
+ */
+function newAnnotations(state) {
+  return state.annotations.annotations.filter(function(ann) {
+    return metadata.isNew(ann);
+  });
+}
+
+/**
+ * Return all loaded annotations that are highlights and have not been saved
+ * to the server.
+ */
+function newHighlights(state) {
+  return state.annotations.annotations.filter(function(ann) {
+    return metadata.isNew(ann) && metadata.isHighlight(ann);
+  });
+}
```

Comment:
I think it would probably be a good idea to use `createSelector` to optimize `newAnnotations` and `newHighlights` and reduce the amount of work that is done when actions that don't touch annotations at all pass through the store.

Reply:
Great suggestion!